### DISCUSSION TITLE
Server script support

### DISCUSSION THREAD

**Discussion by Veritius** [FakeFishGames/Barotrauma#6081] [Ideas and suggestions]
Dedicated servers supporting scripts. Basically, when the server starts it looks for scripts in a specific folder, and those scripts will alter the game in some way, such as adding a command or making someone explode.
These scripts would almost certainly have to run server-side, or have limited interaction with clients, for security purposes.

This would likely be a very large undertaking for the devs, but it would certainly allow for a lot more flexibility for more unconventional approaches to gameplay.

**Comment by evilfactory** [comment]
I already have something similar that adds lua scripting to barotrauma on the server-side, it allows you to listen for chat messages, execute commands and other things, although i wanted it to be client-side too but apparently the devs removed the Executable tag from core content packages. https://steamcommunity.com/sharedfiles/filedetails/?id=2559634234

**Reply by neirenoir** [reply]
Is there any reason as to why Lua scripting has not be merged upstream? I could understand why adding C# modding may be a terrible idea, but your fork sandboxes Lua just right.
